From b28f7bd9bba6f65258da48955154794f466b4104 Mon Sep 17 00:00:00 2001
From: Bruce Ashfield <bruce.ashfield@gmail.com>
Date: Thu, 30 Dec 2021 15:23:59 +0200
Subject: [PATCH] btrfs-progs: include linux/const.h to fix build with 5.12+
 headers

btrfs-tools compile fails with mips, musl and 5.12+ headers.

The definition of __ALIGN_KERNEL has moved in 5.12+ kernels, so we
add an explicit include of const.h to pickup the macro:

  | make: *** [Makefile:595: mkfs.btrfs] Error 1
  | make: *** Waiting for unfinished jobs....
  | libbtrfs.a(volumes.o): in function `dev_extent_search_start':
  | /usr/src/debug/btrfs-tools/5.12.1-r0/git/kernel-shared/volumes.c:464: undefined reference to `__ALIGN_KERNEL'
  | collect2: error: ld returned 1 exit status

This is safe for older kernel's as well, since the header still
exists, and is valid to include.

Signed-off-by: Bruce Ashfield <bruce.ashfield@gmail.com>
[remove invalid OE Upstream-status]
Signed-off-by: Stijn Tintel <stijn@linux-ipv6.be>
Signed-off-by: David Sterba <dsterba@suse.com>
[Retrieved from:
https://github.com/kdave/btrfs-progs/commit/b28f7bd9bba6f65258da48955154794f466b4104]
Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
---
 kerncompat.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kerncompat.h b/kerncompat.h
index df167fe6c..2503d1afa 100644
--- a/kerncompat.h
+++ b/kerncompat.h
@@ -29,6 +29,7 @@
 #include <stddef.h>
 #include <linux/types.h>
 #include <linux/kernel.h>
+#include <linux/const.h>
 #include <stdint.h>
 
 #include <features.h>
